### PR TITLE
Another stripe branch

### PR DESCRIPTION
**Pull request by obaidmm**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -9,6 +9,8 @@
       "version": "0.1.0",
       "dependencies": {
         "@fortawesome/fontawesome-free": "^6.5.1",
+        "@stripe/react-stripe-js": "^2.5.0",
+        "@stripe/stripe-js": "^3.0.0",
```
Comment (obaidmm) [inline package-lock.json]: Adding stripe dependecies

File: .gitignore [hunk 2]
```diff
@@ -1 +1,10 @@
 node_modules
+
+.env
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
```
Comment (obaidmm) [inline .gitignore]: adding additional ignores

File: src/App.js [hunk 3]
```diff
@@ -4,6 +4,10 @@ import TeamsPage from "./Pages/TeamsPage/TeamsPage";
 import SignInPage from "./Pages/SignInPage/SignIn";
 import Settings from "./Pages/SettingsPage/Settings"
 
+import Payment from "./Pages/StripePage/Payment";
+import CheckoutForm from "./Pages/StripePage/CheckoutForm"; 
+import CompletionPage from "./Pages/StripePage/Completion";
```
Comment (obaidmm) [inline src/App.js]: importing files

File: src/App.js [hunk 4]
```diff
@@ -18,20 +22,33 @@ function App() {
...
+    {page === "Homepage" && <HomePage links1={goTeamsPage} links2={goSignInPage} links3={settings} />}
+    {page === "Teams" && <TeamsPage returnHome={returnHome} />}
+    {page === "SignIn" && <SignInPage returnHome={returnHome} />}
+    {page === "Settings" && <Settings returnHome={returnHome} />}
+    {page === "Payment" && <Payment />} // Add a conditional render for Payment
+    {page === "Checkout" && <CheckoutForm />} // Add a conditional render for CheckoutForm
+    {page === "Completion" && <CompletionPage />} // Add a conditional render for CompletionPage
+    {/* Continue with conditional rendering for other pages */}
+  </div>
   );
```
Comment (obaidmm) [inline src/App.js]: importing all the new files for stripe

File: src/App.js [hunk 5]
```diff
@@ -18,20 +22,33 @@ function App() {
     setPage("SignIn")
     
   };
+  
+  const goPayment = () => setPage("Payment"); // Add a function to go to the payment page
+
```
Comment (obaidmm) [inline src/App.js]: directing towards the checkout page

File: src/Pages/TeamsPage/TeamsPage.js [hunk 6]
```diff
@@ -51,73 +145,73 @@ const TeamsPage = ({returnHome}) => {
 
     var [page, setPage] = useState('Account');
 
-    return(
-            <div className="flex bg-white overflow-auto ">
+    return (
```
Comment (obaidmm) [inline src/Pages/TeamsPage/TeamsPage.js]: this can all just be added

File: src/Pages/TeamsPage/TeamsPage.js [hunk 7]
```diff
@@ -2,40 +2,134 @@
...
+        { Poster: "N/A", Name: "Ganesh", id: 2, Description: "Hi my name is Ganesh. I'm a hard working student from Western", Skills: ['Hard working', 'Collaborative', 'Creative', 'Innovative', 'Leader'] },
+        { Poster: "N/A", Name: "Rad", id: 3, Description: "Hi my name is Rad. I never show up", Skills: ['Hard working', 'Collaborative', 'Creative', 'Innovative', 'Leader'] },
+        { Poster: "N/A", Name: "Shivam", id: 3, Description: "Hi my name is Shivam. I'm a hard working student from Western", Skills: ['Hard working', 'Collaborative', 'Creative', 'Innovative', 'Leader'] },
+        { Poster: "N/A", Name: "Ade", id: 4, Description: "Hi my name is Ade. I'm a hard working student from Western", Skills: ['Hard working', 'Collaborative', 'Creative', 'Innovative', 'Leader'] },
+        { Poster: "N/A", Name: "Joann", id: 4, Description: "Hi my name is Joann. I'm a hard working student from Western", Skills: ['Hard working', 'Collaborative', 'Creative', 'Innovative', 'Leader'] }
```
Comment (obaidmm) [inline src/Pages/TeamsPage/TeamsPage.js]: again just take the new one

File: src/Pages/TeamsPage/TeamsPage.js [hunk 8]
```diff
@@ -2,40 +2,134 @@
...
+
+        const stripe = await getStripe();
+        const { error } = await stripe.redirectToCheckout(checkoutOptions);
+        console.log("Stripe checkout error", error);
+
+        if (error) setStripeError(error.message);
+        setLoading(false);
+    };
+
+    if (stripeError) alert(stripeError);
```
Comment (obaidmm) [inline src/Pages/TeamsPage/TeamsPage.js]: I would need all of this

File: src/Pages/TeamsPage/TeamsPage.js [hunk 9]
```diff
@@ -2,40 +2,134 @@
 import { HiArrowSmallLeft } from "react-icons/hi2";
 
 //Importing neccesary libraries
+import React, { useState } from 'react';
+import { loadStripe } from '@stripe/stripe-js';
+
```
Comment (obaidmm) [inline src/Pages/TeamsPage/TeamsPage.js]: this is just libraries for stripe

File: src/Pages/HomePage/HomePage.jsx [hunk 10]
```diff
@@ -13,7 +13,7 @@ const HomePage = ({links1, links2, links3}) =>{
             <Navbar links1 = {links1} links2 = {links2} links3={links3}/>
             <Hero links2 = {links2}/>
             <Analytics links1 = {links1}/>
-            <div className='fixed top-[30%] right-5'>
+            <div className='fixed top-[30%] right-0 z-5'>
```
Comment (obaidmm) [inline src/Pages/HomePage/HomePage.jsx]: just additional things for css

File: public/index.html [hunk 11]
```diff
@@ -24,6 +24,8 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
+    <script src="https://js.stripe.com/v3/"></script>
+
```
Comment (obaidmm) [inline public/index.html]: needed for stripe

**Review by ColeBranston — APPROVED** [review]

**Review by ColeBranston — APPROVED** [review]

**Review by ColeBranston — APPROVED** [review]

**Pull request closed without merging**